File: 3 КУРС/2 семестр/Java/VoiceAssistent/MessageViewHolder.java
package com.example.voiceassistent;


import android.view.View;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.text.DateFormat;
import java.text.SimpleDateFormat;

public class MessageViewHolder extends RecyclerView.ViewHolder {

    protected TextView messageText;
    protected TextView messageDate;

    public MessageViewHolder(@NonNull View itemView) {
        super(itemView);
        messageText = itemView.findViewById(R.id.messageTextView);
        messageDate = itemView.findViewById(R.id.messageDateView);
    }
    public void bind(Message message) {
        messageText.setText(message.text);
        DateFormat fmt = new SimpleDateFormat("k':'m");
        messageDate.setText(fmt.format(message.date));
    }

}

File: 3 КУРС/2 семестр/Учебная практика/asd/JavaProects/app/src/main/java/com/example/voiceassistent/MessageViewHolder.kt
package com.example.voiceassistent

import android.annotation.SuppressLint
import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import java.text.DateFormat
import java.text.SimpleDateFormat

class MessageViewHolder(itemView: View) : ViewHolder(itemView) {
    protected var messageText: TextView
    protected var messageDate: TextView
    fun bind(message: Message) {
        messageText.text = message.text
        @SuppressLint("SimpleDateFormat") val fmt: DateFormat = SimpleDateFormat("HH:mm")
        messageDate.text = fmt.format(message.date)
    }

    init {
        messageText = itemView.findViewById(R.id.messageTextView)
        messageDate = itemView.findViewById(R.id.messageDateView)
    }
}